pytorch_lightning/callbacks/progress.py

```diff
@@ -0,0 +1,371 @@
+"""
+Progress Bars
+=============
+
+Use or override one of the progress bar callbacks.
+
+"""
+import sys
+from typing import Optional
+
+from tqdm.auto import tqdm
+
+from pytorch_lightning.callbacks import Callback
+
+
+class ProgressBarBase(Callback):
+    r"""
+    The base class for progress bars in Lightning. It is a :class:`~pytorch_lightning.callbacks.Callback`
+    that keeps track of the batch progress in the :class:`~pytorch_lightning.trainer.trainer.Trainer`.
+    You should implement your highly custom progress bars with this as the base class.
+
+    Example::
+
+        class LitProgressBar(ProgressBarBase):
+
+            def __init__(self):
+                super().__init__()  # don't forget this :)
+                self.enabled = True
+
+            def disable(self):
+                self.enabled = False
+
+            def on_batch_end(self, trainer, pl_module):
+                super().on_batch_end(trainer, pl_module)  # don't forget this :)
+                percent = (self.train_batch_idx / self.total_train_batches) * 100
+                sys.stdout.flush()
+                sys.stdout.write(f'{percent:.01f} percent complete \r')
+
+        bar = LitProgressBar()
+        trainer = Trainer(callbacks=[bar])
+
+    """
+    def __init__(self):
+
+        self._trainer = None
+        self._train_batch_idx = 0
+        self._val_batch_idx = 0
+        self._test_batch_idx = 0
+
+    @property
+    def trainer(self):
+        return self._trainer
+
+    @property
+    def train_batch_idx(self) -> int:
+        """
+        The current batch index being processed during training.
+        Use this to update your progress bar.
+        """
+        return self._train_batch_idx
+
+    @property
+    def val_batch_idx(self) -> int:
+        """
+        The current batch index being processed during validation.
+        Use this to update your progress bar.
+        """
+        return self._val_batch_idx
+
+    @property
+    def test_batch_idx(self) -> int:
+        """
+        The current batch index being processed during testing.
+        Use this to update your progress bar.
+        """
+        return self._test_batch_idx
+
+    @property
+    def total_train_batches(self) -> Optional[int]:
+        """
+        The total number of training batches during training, which may change from epoch to epoch.
+        Use this to set the total number of iterations in the progress bar. Can return ``inf`` if the
+        training dataloader is of infinite size.
+        """
+        if self.trainer.fast_dev_run:
+            total_train_batches = 1
+        else:
+            total_train_batches = self.trainer.num_training_batches
```

Comment:
```suggestion
        total_train_batches = 1 if self.trainer.fast_dev_run else self.trainer.num_training_batches
```
just for luck of test coverage

Reply:
this influences test coverage?

Reply:
well you have to have tests for both cases.. and if you squash it to one to is used always lol